bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -330,6 +331,28 @@ private BigtableOptions resolveLegacyOptions(BigtableOptions options) throws IOE
     return options;
   }
 
+
+  /**
+   * Return options with legacy input options, if any, resolved into currently supported options.
+   */
+  public synchronized BigtableClusterName getClusterName() throws IOException {
+    if (options.getClusterName() == null) {
```

Comment:
This is a public method; can you give me an example how it should be used?

I thought what need to done in the constructor is:
if (instance ID provided) {
  cluster ID = get single cluster ID from instance ID.
  check cluster ID is the same as input cluster ID if provided, otherwise set cluster ID in options.
} else {
  instance ID = get single instance ID by cluster ID.
  set instance ID in options.
}